### DISCUSSION TITLE
useScrollbar

### DISCUSSION THREAD

**Discussion by bstro** [adobe/react-spectrum#4122] [Ideas]
I haven't seen anything like this in react-spectrum/aria after searching on google/issues/discussion, forgive me if this is a dupe of some kind.

I'm needing a custom scrollbar that is ideally just as accessible as a native scrollbar and I'd rather write it in React rather than finagle with restyling the native scrollbar itself.

I'm noticing that there is an aria role for this already: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/scrollbar_role

Curious if this is something that has already been considered by the Spectrum team.

**Comment by reidbarber** [comment]
I don't think we have plans to add a custom scrollbar to React Spectrum since we rely on the native one.

That being said, I can bring it up with the team on whether we would be open to accepting a contribution of this to React Aria for this.

I think you could borrow a lot of code from [`useSlider`](https://github.com/adobe/react-spectrum/blob/main/packages/%40react-aria/slider/src/useSlider.ts) to implement your own.

**Reply by bstro** [reply]
Cool, that could be an interesting project to work on, I might be down!
